### PR TITLE
Depencdencies layout incidentfield

### PR DESCRIPTION
**Pull request by teizenman**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
Related: https://github.com/demisto/etc/issues/24610

## Description
Added the fields to id_set

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/common/update_id_set.py [hunk 1]
```diff
@@ -287,11 +347,14 @@ def get_layout_data(path):
     toversion = json_data.get('toVersion')
     kind = json_data.get('kind')
     pack = get_pack_name(path)
+    incident_indicator_types_dependency = {id_}
```
Comment (ronykoz) [inline demisto_sdk/commands/common/update_id_set.py:350]: the id of the layout is the type?

File: demisto_sdk/commands/common/update_id_set.py [hunk 2]
```diff
@@ -302,6 +365,53 @@ def get_layout_data(path):
...
+    all_associated_types = set()
+    all_scripts = set()
+
+    associated_types = json_data.get('associatedTypes')
+    if associated_types:
+        all_associated_types = set(associated_types)
+
+    system_associated_types = json_data.get('systemAssociatedTypes')
+    if system_associated_types:
+        all_associated_types = all_associated_types.union(set(system_associated_types))
```
Comment (ronykoz) [inline demisto_sdk/commands/common/update_id_set.py:393]: what if i have all here?

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 3]
```diff
@@ -292,6 +292,82 @@ def _collect_playbooks_dependencies(pack_playbooks, id_set):
...
+        for layout in pack_layouts:
+            layout_data = next(iter(layout.values()))
+
+            related_incident_and_indicator_types = layout_data.get('incident_and_indicator_types', [])
+            packs_found_from_incident_indicator_types = PackDependencies._search_packs_by_items_names(
+                related_incident_and_indicator_types, id_set['IncidentTypes'] + id_set['IndicatorTypes'])
+
+            if packs_found_from_incident_indicator_types:
+                pack_dependencies_data = PackDependencies. \
+                    _label_as_mandatory(packs_found_from_incident_indicator_types)
```
Comment (IkaDemisto) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:319]: Do we want to label all found results as mandatory dependency or there can be cases where incident type/indicator type are generic? Similar to reputations commands in integrations

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 4]
```diff
@@ -292,6 +292,82 @@ def _collect_playbooks_dependencies(pack_playbooks, id_set):
...
+                dependencies_packs.update(pack_dependencies_data)
+
+            related_incident_and_indicator_fields = layout_data.get('incident_and_indicator_fields', [])
```
Comment (IkaDemisto) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:322]: May be confusing. Why incident and indicator fields are in same list in id-set?

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 5]
```diff
@@ -292,6 +292,82 @@ def _collect_playbooks_dependencies(pack_playbooks, id_set):
...
+            packs_found_from_incident_indicator_fields = PackDependencies._search_packs_by_items_names(
+                related_incident_and_indicator_fields, id_set['IncidentFields'] + id_set['IndicatorFields'])
+
+            if packs_found_from_incident_indicator_fields:
+                pack_dependencies_data = PackDependencies. \
+                    _label_as_mandatory(packs_found_from_incident_indicator_fields)
```
Comment (IkaDemisto) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:328]: Same question here regarding setting mandatory/optional dependency

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 6]
```diff
@@ -292,6 +292,82 @@ def _collect_playbooks_dependencies(pack_playbooks, id_set):
...
+            related_incident_types = incident_field_data.get('incident_types', [])
+            packs_found_from_incident_types = PackDependencies._search_packs_by_items_names(
+                related_incident_types, id_set['IncidentTypes'])
+
+            if packs_found_from_incident_types:
+                pack_dependencies_data = PackDependencies. \
+                    _label_as_mandatory(packs_found_from_incident_types)
+                dependencies_packs.update(pack_dependencies_data)
+
+            related_scripts = incident_field_data.get('scripts', [])
```
Comment (IkaDemisto) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:360]: Can you explain more on it? I don't find example in generated id-set test file.

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 7]
```diff
@@ -321,10 +401,14 @@ def _find_pack_dependencies(pack_id, id_set):
...
-        pack_dependencies = scripts_dependencies | playbooks_dependencies
+        pack_items = PackDependencies._collect_pack_items(pack_id, id_set)
+        scripts_dependencies = PackDependencies._collect_scripts_dependencies(pack_items['scripts'], id_set)
+        playbooks_dependencies = PackDependencies._collect_playbooks_dependencies(pack_items['playbooks'], id_set)
+        layouts_dependencies = PackDependencies._collect_layouts_dependencies(pack_items['layouts'], id_set)
+        incidents_fields_dependencies = PackDependencies. \
+            _collect_incidents_fields_dependencies(pack_items['incidents_fields'], id_set)
+        pack_dependencies = scripts_dependencies | playbooks_dependencies | layouts_dependencies | \
+            incidents_fields_dependencies
         # todo check if need to collect dependencies from other content items
```
Comment (IkaDemisto) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:413]: https://github.com/demisto/demisto-sdk/blob/333b2e865e9888d2b35c8751311dbfa09a775bbc/demisto_sdk/tests/test_files/id_set/id_set.json is real id-set that was generated according to your changes, correct?

**Comment by IkaDemisto** [comment]
@teizenman @ronykoz will be included in upcoming sdk release?

**Comment by anara123** [comment]
@teizenman please upload here `id_set.json` after your changes

**Comment by teizenman** [comment]
@anara123 Attached

[id_set.json.zip](https://github.com/demisto/demisto-sdk/files/4780958/id_set.json.zip)

**Pull request merged**